packages/portfolio-contract/tools/plan-solve.ts

```diff
@@ -381,24 +338,30 @@ export const rebalanceMinCostFlowSteps = async (
   let lastChain: string | undefined;
 
   while (pendingFlows.size) {
-    const candidates = [...pendingFlows.values()].filter(
-      f => (supplies.get(f.edge.src) || 0) >= f.flow,
-    );
+    // Find flows that can be executed based on current supplies.
+    // Use tolerance to avoid spurious deadlocks from:
+    // - Floating-point rounding accumulated during supply tracking
+    // - Integer division rounding in target allocation computations
+    // Apply both absolute and relative tolerance - take the larger of the two.
+    const candidates = [...pendingFlows.values()].filter(f => {
```

Comment:
The change to candidates selection here does not seem required for the jsLPSolver.

Reply:
Reverted.